Source/Core/Core/Scripting/Documentation/PythonRawAPIDocumentation.txt

```diff
@@ -0,0 +1,895 @@
+		  GraphicsAPI.addButton("Click Me!", 42, onClickFunction, 200.0, 100.0) ==> After this, a button exists which says "Click Me!". If the user clicks this button, it will call the function onClickFunction
+
+	void pressButton(s64 uniqueButtonId):
+		- Calls the callback function associated with the button which has a unique button ID of uniqueButtonId
+		  GraphicsAPI.pressButton(42) ==> If the addButton() call in the example above was run right before this, then this function would cause the onClickFunction to be called.
+
+	void newLine(float verticalSpacing):
+		- Pushes the screen cursor down by the amount specified by verticalSpacing. Only affects checkboxes, radio buttons, text boxes, and buttons.
+		  GraphicsAPI.newLine(15.0) ==> Advances the cursor by 15.0 pixels
+
+	void beginWindow(String windowName):
+		- Creates a new window with the name/label specified by windowName. This window will be the currently opened window until another call to beginWindow() or endWindow() is made. The first call to this function will create a window, and any subsequent calls to beginWindow() made before the window has been closed will create a nested sub-window inside of the currently opened window. There must be an equal number of calls to beginWindow() and endWindow() on each frame.
+		  GraphicsAPI.beginWindow("My Window") ==> Displays a window on screen with the title "My Window"
+
+	void endWindow():
+		- Finalizes/finishes the currently opened window. There must be an equal number of calls to beginWindow() and endWindow() on each frame.
+		  GraphicsAPI.endWindow() ==> After this, the next time addButton is called, it won't add the button to the window which was previously the current window (since that is now done being written to).
+
+--------------------------------------------------------------------------------------------------------------------------------------------
+InstructionStepAPI class:
+
+Calling any of the functions below will throw an exception if it's not called from an OnInstructionHit callback, an OnMemoryAddressReadFrom callback, or an OnMemoryAddressWrittenTo callback. The lone exception to this is the getInstructionFromAddress() function, which can be called from anywhere).
+
+	String getInstructionFromAddress(u32 addressVal):
+		- Returns a human-readable String representation of the instruction at the address specified by addressVal.
+		  InstructionStepAPI.getInstructionFromAddress(0X84445578) ==> Returns a representation of the instruction at address 0X84445578. Ex. "addi	sp, sp, 8"
```

Comment:
Are Memory Addresses Masked? For example in Dolphin Memory Engine (Which I've personally used a lot), the Address `0x8000_0004` gets masked to `0x0000_0004`. Since that is more correct to the GC's Address Space.

Reply:
I use the default value for RequestedAddressSpace in each function which is shown by this line: "RequestedAddressSpace space = RequestedAddressSpace::Effective"

This is in contrast to using Physical or Virtual addresses. In this case, memory addresses act according to whether or not MMU is turned on.

Reply:
I'm still not 100% sure what you mean here. I'll use Screenshot from the Debugging UI to help clarify what I mean.
![image](https://github.com/dolphin-emu/dolphin/assets/25422785/4be33d35-31ab-4036-bdd5-54766c4d659a)

We can see the Game's Header begins at `0x8000_0000` in relation to the Memory Addresses on the left, so if I wanted to get the header, would I just do `0x8000_0000`, or `0x0000_0000`? The first one mirrors Dolphin, however the second one is the effective address, as it starts on the zeroth Byte of RAM.

Reply:
Should be 0X8000_0000

Reply:
> The first one mirrors Dolphin, however the second one is the effective address, as it starts on the zeroth Byte of RAM.

No, it is the physical address.